otelcollector/telegraf/telegraf-prometheus-collector-ds.conf

```diff
@@ -0,0 +1,206 @@
+  ## Each plugin will sleep for a random time within jitter before collecting.
+  ## This can be used to avoid many plugins querying things like sysfs at the
+  ## same time, which can have a measurable effect on the system.
+  collection_jitter = "0s"
+
+  ## Default flushing interval for all outputs. You shouldn't set this below
+  ## interval. Maximum flush_interval will be flush_interval + flush_jitter
+  flush_interval = "15s"
+  ## Jitter the flush interval by a random amount. This is primarily to avoid
+  ## large write spikes for users running a large number of telegraf instances.
+  ## ie, a jitter of 5s and interval 10s means flushes will happen every 10-15s
+  flush_jitter = "0s"
+
+  ## By default or when set to "0s", precision will be set to the same
+  ## timestamp order as the collection interval, with the maximum being 1s.
+  ##   ie, when interval = "10s", precision will be "1s"
+  ##       when interval = "250ms", precision will be "1ms"
+  ## Precision will NOT be used for service inputs. It is up to each individual
+  ## service input to set the timestamp at the appropriate precision.
+  ## Valid time units are "ns", "us" (or "µs"), "ms", "s".
+  precision = ""
+
+  ## Logging configuration:
+  ## Run telegraf with debug log messages.
+  debug = false
+  ## Run telegraf in quiet mode (error log messages only).
+  quiet = true
+  ## Specify the log file name. The empty string means to log to stderr.
+  logfile = ""
+
+  ## Override default hostname, if empty use os.Hostname()
+  #hostname = "placeholder_hostname"
+  ## If set to true, do no set the "host" tag in the telegraf agent.
+  omit_hostname = true
+
+
+###############################################################################
+#                            OUTPUT PLUGINS                                   #
+###############################################################################
+
+[[outputs.application_insights]]
+  ## Instrumentation key of the Application Insights resource.
+  instrumentation_key = "$TELEMETRY_APPLICATIONINSIGHTS_KEY"
+
+  ## Timeout for closing (default: 5s).
+  # timeout = "5s"
+
+  ## Enable additional diagnostic logging.
+  # enable_diagnostic_logging = false
+
+###############################################################################
+#                            PROCESSOR PLUGINS                                #
+###############################################################################
+[[processors.converter]]
+  [processors.converter.fields]
+    float = ["*"]
+
+# Transforms tag and field values as well as measurement, tag and field names with regex pattern
+[[processors.regex]]
+  namepass = ["opentelemetry_allocator_targets"]
+
+  # Tag and field conversions defined in a separate sub-tables
+  [[processors.regex.tags]]
+    ## Tag to change, "*" will change every tag
+    key = "job_name"
+    ## Regular expression to match on a tag value
+    pattern = '(^(podmonitor|servicemonitor)\/.*)|(.*)'
```

Comment:
@gracewehner - I was double checking the regex and looks like this seems to match any job name that has podmonitor in it? ex - anynamepodmonitor. Should we be more specific about the job names to match what TA generates?

Reply:
Thanks Rashmi, yes so we want all job names to match so that they are replaced/overwritten and we aren't collecting the name. So we're only putting podmonitor or servicemonitor if it matches group 2, which has "podmonitor/*" will be replaced as podmonitorJob since it matches that it starts with "podmonitor/". But anynamepodmonitor doesn't match group 2 but just the right-hand side of the | (.*), so it's replaced as just 'Job' to signify it's a configmap name. So ${2} will be empty. Telegraf just won't allow empty string replacements, so I've added it to always suffix with Job

Reply:
sounds good, thanks Grace